app/models/solidus_acima/gateway.rb

```diff
@@ -67,18 +67,23 @@ def void(response_code, options)
 
     def credit(_amount, response_code, options)
       payment_source = options[:originator].payment.source
+      lease_id = payment_source.lease_id
 
-      url = "#{api_url}/contracts/#{payment_source.lease_id}/termination"
-      response = HTTParty.post(url, headers: v2_headers)
+      if acima_payment_captured?(lease_id)
+        url = "#{api_url}/contracts/#{lease_id}/termination"
+        response = HTTParty.post(url, headers: v2_headers)
 
-      raise 'Acima Server Response Error: Did not get correct response code' unless response.success?
+        raise 'Acima Server Response Error: Did not get correct response code' unless response.success?
 
-      ActiveMerchant::Billing::Response.new(
-        true,
-        'Transaction credited',
-        response || {},
-        authorization: response_code
-      )
+        ActiveMerchant::Billing::Response.new(
+          true,
+          'Transaction credited',
+          response || {},
+          authorization: response_code
+        )
+      else
+        void(response_code, { originator: options[:originator].payment })
```

Comment:
it looks a little bit strange to me making a void inside a credit action, without notifying the user. Maybe it would be better to disable the credit button when [can_credit?](https://github.com/solidusio/solidus/blob/e878076f2ed670d07654ab6293a16588743f2fa6/core/spec/support/concerns/payment_source.rb#L36) returns false

Reply:
I will close this PR and create a new branch with that logic instead